extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/PathParameterValueGettersGenerator.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.resteasy.reactive.links.deployment;
+
+import java.util.Set;
+import java.util.function.BiFunction;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.IndexView;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+import io.quarkus.deployment.util.AsmUtil;
+import io.quarkus.gizmo.DescriptorUtils;
+import io.quarkus.gizmo.Gizmo;
+import io.quarkus.resteasy.reactive.links.runtime.PathParameterInfo;
+
+public class PathParameterValueGettersGenerator extends ClassVisitor {
```

Comment:
Let's have some comments on what exactly this does. ASM code is always hard to figure out, so let's at least have comments on what the intended functionality is

Reply:
Sure, I'll explain it once I have the extension finalised. But yes, it generates getters for the fields that are needed for path parameter values.